apps/angular/crud/src/stores/app.store.ts

```diff
@@ -0,0 +1,93 @@
+import { Injectable } from '@angular/core';
+import { ComponentStore } from '@ngrx/component-store';
+import { concatMap, exhaustMap, Observable, tap } from 'rxjs';
+import { TodoHttpService } from '../http/todo-http.service';
+import { TodoItem } from '../models/todo';
+
+export interface AppState {
+  isAppProcessing: boolean;
+  todos: TodoItem[];
+  processingTodos: Set<number>;
+}
+
+@Injectable()
+export class AppStore extends ComponentStore<AppState> {
+  constructor(private todoHttpService: TodoHttpService) {
+    super({
+      isAppProcessing: false,
+      todos: [],
+      processingTodos: new Set(),
+    });
+  }
+
+  readonly isAppProcessing$ = this.select((state) => state.isAppProcessing);
+  readonly todos$ = this.select((state) => state.todos);
+
+  setTodoItemProcessing(id: number, flag: boolean) {
+    this.patchState((state) => {
+      if (flag) {
+        state.processingTodos.add(id);
+      } else {
+        state.processingTodos.delete(id);
+      }
+      return state;
+    });
+  }
+
+  isTodoItemProcessing(id: number): boolean {
+    return this.state().processingTodos.has(id);
+  }
+
+  readonly getTodoItems = this.effect((trigger$) => {
+    this.setState((state) => ({ ...state, isAppProcessing: true }));
```

Comment:
you should add this inside the steam with the tap operator. effect is mostly a pipe